app/src/apkinstaller.h

```diff
@@ -0,0 +1,55 @@
+#ifndef APK_INSTALLER_H
+#define APK_INSTALLER_H
+
+#define APK_QUEUE_SIZE 16
+#define MAX_FILENAME_SIZE 1024
+
+
+#include "apkinstaller.h"
+
+#include <SDL2/SDL_mutex.h>
+#include <SDL2/SDL_stdinc.h>
+#include <SDL2/SDL_thread.h>
+#include "command.h"
+
+// NOTE(AdoPi) apk_queue and control_event can use a generic queue
+
+struct apk_queue {
+    char data[APK_QUEUE_SIZE][MAX_FILENAME_SIZE];
```

Comment:
That's wasteful and unsafe. You should use allocated strings (with `SDL_strdup`) instead (and use `char *data[APK_QUEUE_SIZE]` here).